### PR TITLE
High Performance clock

### PR DESCRIPTION
**Pull request by IceDragon200**
Currently only the linux implementation is in place, this replaces the need for glfw in the system module for time keeping.

@archSeer Can you implement the Mac specific one, OSX stuff scares me. :(

### CODE REVIEW

**Comment by IceDragon200** [comment]
This PR requires #17 to build correctly

**Comment by archSeer** [comment]
I honestly don't think this is a good idea. GLFW already uses mach_time, posix_time or win32_time. It just makes it a bigger hassle for us to maintain code across platforms. What is the reasoning behind this?

https://github.com/glfw/glfw/search?utf8=%E2%9C%93&q=_glfwPlatformGetTime&type=Code

**Comment by IceDragon200** [comment]
And you need to initialize GLFW, which requires a display driver present in order to get time, not to mention this forces the system module, whose sole purpose is to define core data classes and utilities to carry around the weight of a windowing gem, when all it wants from it is time.

```ruby
> GLFW.time
(mirb):2: 65537: The GLFW library is not initialized (GLFWError)
``` 

The `GLFW.time` here is just a quick wrapper around:
```c++
mrb_float_value(mrb, glfwGetTime());
```

So, we can ignore this, keep `glfw` as a dependency of the system module, and then I disable the clock test to have tests "pass".

Besides when you implement it once, you won't have to change it again unless someone magically comes out with a better time api, for a target OS.


tl;dr IMPLEMENT OR CONTINUE TO HAVE TEST FAIL OR JUST DISABLE THE CLOCK TEST

**Comment by archSeer** [comment]
Closing this for now too.
